### PR TITLE
Enhance build system for users

### PR DESCRIPTION
**Pull request by lherman-cs**
Changes:
* Add install steps
* Add pc files

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMake/Utilities.cmake [hunk 1]
```diff
@@ -36,30 +36,30 @@ function(build_dependency lib_name)
...
   set(build_args ${ARGN})
 
-  file(REMOVE_RECURSE ${KINESIS_VIDEO_OPEN_SOURCE_SRC}/lib${lib_name})
+  file(REMOVE_RECURSE ${OPEN_SRC_INSTALL_PREFIX}/lib${lib_name})
 
   # build library
   configure_file(
-    ${CMAKE_SOURCE_DIR}/CMake/Dependencies/lib${lib_name}-CMakeLists.txt
-    ${KINESIS_VIDEO_OPEN_SOURCE_SRC}/lib${lib_name}/CMakeLists.txt COPYONLY)
+    ./CMake/Dependencies/lib${lib_name}-CMakeLists.txt
```
Comment (MushMal) [inline CMake/Utilities.cmake:43]: Does cmake have a pre-configured variable for this path?

File: CMakeLists.txt [hunk 2]
```diff
@@ -112,19 +113,49 @@ if(UNIX AND NOT APPLE)
...
+
 add_library(kvspicUtils STATIC ${PIC_UTILS_SOURCE_FILES})
 target_link_libraries(kvspicUtils ${CMAKE_DL_LIBS} Threads::Threads)
 if(UNIX AND NOT APPLE)
   # rt needed for clock_gettime
   target_link_libraries(kvspicUtils rt)
 endif()
 
+install(
+  TARGETS kvspic kvspicClient kvspicState kvspicUtils
```
Comment (MushMal) [inline CMakeLists.txt:142]: Will this properly create a directory per project for the include files? I assume so as we would end up with name conflicts as all the public includes are named Include.h

File: libkvspic.pc.cmake [hunk 3]
```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+includedir=${prefix}/include
```
Comment (MushMal) [inline libkvspic.pc.cmake:3]: Oh, there it is

File: CMake/Utilities.cmake (hunk 1 above)
Comment (lherman-cs) [reply CMake/Utilities.cmake:43]: AFAIK, this path is custom to us, so I don't think CMAKE have a pre-configure variable for this.

File: CMakeLists.txt (hunk 2 above)
Comment (lherman-cs) [reply CMakeLists.txt:142]: Yep, the prefix is configurable using `CMAKE_INSTALL_PREFIX`

**Pull request merged**